contracts/kleros/KlerosLiquid.sol

```diff
@@ -0,0 +1,752 @@
+
+    /** @dev Notifies the controller about a token transfer allowing the controller to react if desired.
+     *  @param _from The origin of the transfer.
+     *  @param _to The destination of the transfer.
+     *  @param _amount The amount of the transfer.
+     *  @return Wether the operation should be allowed or not.
+     */
+    function onTransfer(address _from, address _to, uint _amount) public returns(bool allowed) {
+        uint _newBalance = pinakion.balanceOf(_from) - _amount;
+        require(_newBalance >= stakeOf(bytes32(0), msg.sender), "Cannot transfer an amount that would make balance less than stake.");
+        require(_newBalance >= jurors[_from].atStake, "Cannot transfer an amount that would make balance less than locked stake.");
+        allowed = true;
+    }
+
+    /** @dev Notifies the controller about an approval allowing the controller to react if desired.
+     *  @param _owner The address that calls `approve()`.
+     *  @param _spender The spender in the `approve()` call.
+     *  @param _amount The amount in the `approve()` call.
+     *  @return Wether the operation should be allowed or not.
+     */
+    function onApprove(address _owner, address _spender, uint _amount) public returns(bool allowed) { allowed = true; }
+
+    /* Public Views */
+
+    /** @dev Implement original `arbitrationCost()` from standard for compilation.
+     *  @param _extraData Additional info about the dispute.
+     *  @return The cost.
+     */
+    function arbitrationCost(bytes _extraData) public view returns(uint cost) {
+        cost = NON_PAYABLE_AMOUNT;
+    }
+
+    /** @dev Get the cost of arbitration in a specified subcourt.
+     *  @param _subcourtID The ID of the subcourt.
+     *  @param _extraData Additional info about the dispute.
+     *  @return The cost.
+     */
+    function arbitrationCost(uint _subcourtID, bytes _extraData) public view returns(uint cost) {
+        cost = courts[_subcourtID].jurorFee * courts[_subcourtID].minJurors;
+    }
+
+    /** @dev Get the cost of appealing a specified dispute.
+     *  @param _disputeID The ID of the dispute.
+     *  @param _extraData Additional info about the appeal.
+     *  @return The cost.
+     */
+    function appealCost(uint _disputeID, bytes _extraData) public view returns(uint cost) {
+        Dispute storage dispute = disputes[_disputeID];
+        uint _lastNumberOfJurors = dispute.votes[dispute.votes.length - 1].length;
+        if (_lastNumberOfJurors >= courts[dispute.subcourtID].jurorsForJump) // Jump to parent subcourt.
+            if (dispute.subcourtID == 0) // Already in the general court.
+                cost = NON_PAYABLE_AMOUNT;
+            else
+                cost = courts[courts[dispute.subcourtID].parent].jurorFee * courts[courts[dispute.subcourtID].parent].minJurors;
+        else // Stay in current subcourt.
+            cost = courts[dispute.subcourtID].jurorFee * ((_lastNumberOfJurors * 2) + 1);
```

Comment:
Cost is `2x+1` and I think it's because there will be `2x+1` jurors. But I couldn't find how drawing `2x+1` juror is enforced. Could you point me where to look please?